### PR TITLE
Feat/feedback redesign

### PR DESCRIPTION
**Pull request by aslakihle**
[AB#427022](https://dev.azure.com/Equinor/4319082b-29ba-4111-a20b-c04f55c7e7c6/_workitems/edit/427022)

### CODE REVIEW

File: src/components/Navigation/TopBar/Help/Feedback/Feedback.const.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export const MAX_FILE_SIZE_BYTES = 1048575;
+
+export const SERVICE_NOW_QUERY_KEY = 'serviceNowIncident';
+export const SLACK_POST_QUERY_KEY = 'slackPostMessage';
+export const SLACK_FILE_QUERY_KEY = 'slackFileUpload';
+
+export const DEFAULT_REQUEST_ERROR_MESSAGE =
+  'There was an error with the request';
+
+export const ONE_HOUR_IN_MS: number = 3600000;
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/Feedback.const.ts:12]: Nitpicky but when writing MS constants IMO it's better do just have the full calculation there: `1000 * 60 * 60`

File: src/components/Navigation/TopBar/Help/Feedback/Feedback.types.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { FileWithPath } from 'react-dropzone';
+
+import { ApiError, ServiceNowIncidentResponse } from '../../../../../api';
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/Feedback.types.ts:3]: Import can be shortened

File: src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/FeedbackForm.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+} from '../Feedback.styles';
+import { FeedbackType } from '../Feedback.types';
+import { useFeedbackContext } from '../hooks/useFeedbackContext';
+import Description from './components/Description';
+import Severity from './components/Severity';
+import Title from './components/Title';
+import UploadFile from './components/UploadFile/UploadFile';
+import Url from './components/Url';
+
+interface FeedbackFormProps {}
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/FeedbackForm.tsx:22]: Unnecessary props type?

File: src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/Description.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { FC, FormEvent, useMemo } from 'react';
+
+import { Typography } from '@equinor/eds-core-react';
+import { tokens } from '@equinor/eds-tokens';
+
+import AmplifyTextField from '../../../../../../Inputs/AmplifyTextField';
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/Description.tsx:6]: Import can be shortened IIRC

File: src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/LockedInputTooltip.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { forwardRef, ReactNode, useMemo } from 'react';
+
+import OptionalTooltip from '../../../../../../DataDisplay/OptionalTooltip';
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/LockedInputTooltip.tsx:3]: Same here 👀

File: src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/Title.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import { useFeedbackContext } from '../../hooks/useFeedbackContext';
+import LockedInputTooltip from './LockedInputTooltip';
+
+import styled from 'styled-components';
+
+const Container = styled.div`
+  grid-column: 1/3;
+`;
+
+interface TitleProps {}
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/Title.tsx:13]: Same with these props

File: src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/UploadFile/UploadFile.tsx [hunk 7]
```diff
@@ -58,18 +54,10 @@ function removeDuplicates(
...
 }
 
-interface UploadFileProps {
-  feedbackContent: FeedbackContentType;
-  updateFeedback: (
-    key: keyof FeedbackContentType,
-    newValue: string | UrgencyOption | FileWithPath[]
-  ) => void;
-}
+interface UploadFileProps {}
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/UploadFile/UploadFile.tsx:57]: 👀

File: src/components/Navigation/TopBar/Help/Feedback/FeedbackInner.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import { FC } from 'react';
+
+import FeedbackForm from './FeedbackForm/FeedbackForm';
+import { useFeedbackContext } from './hooks/useFeedbackContext';
+import ResponsePage from './ResponsePage/ResponsePage';
+
+interface FeedbackInnerProps {}
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/FeedbackInner.tsx:7]: 👀

**Review by mariush2 — APPROVED** [review]

**Review by mariush2 — CHANGES_REQUESTED** [review]
Ment to leave Request changes review 🫠

File: src/components/Navigation/TopBar/Help/Feedback/Feedback.types.ts (hunk 2 above)
Comment (aslakihle) [reply src/components/Navigation/TopBar/Help/Feedback/Feedback.types.ts:3]: I thought there was some issue with using absolute path (src/api) in amplify-components, when getting other things from there.

Comment (mariush2) [reply src/components/Navigation/TopBar/Help/Feedback/Feedback.types.ts:3]: I think I fixed it, you can verify by trying to build and use😎

Comment (aslakihle) [reply src/components/Navigation/TopBar/Help/Feedback/Feedback.types.ts:3]: It does work, nice! 👌

File: src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/UploadFile/UploadFile.tsx [hunk 9]
```diff
@@ -4,13 +4,10 @@ import { FileRejection, FileWithPath } from 'react-dropzone';
...
 import { tokens } from '@equinor/eds-tokens';
 
-import FileUploadArea from 'src/components/Inputs/FileUploadArea';
-import ImageFile from 'src/components/Navigation/TopBar/Help/FeedbackForm/components/ImageFile';
-import { MAX_FILE_SIZE_BYTES } from 'src/components/Navigation/TopBar/Help/FeedbackForm/FeedbackForm.const';
-import {
-  FeedbackContentType,
-  UrgencyOption,
-} from 'src/components/Navigation/TopBar/Help/FeedbackForm/FeedbackForm.types';
+import FileUploadArea from '../../../../../../../Inputs/FileUploadArea';
```
Comment (mariush2) [inline src/components/Navigation/TopBar/Help/Feedback/FeedbackForm/components/UploadFile/UploadFile.tsx:7]: Missed one import on my initial review 🫠

**Review by mariush2 — CHANGES_REQUESTED** [review]

**Review by mariush2 — APPROVED** [review]
God helg🥳

**Pull request merged**